pkg/workloadmeta/collectors/internal/kubeapiserver/deployments_test.go

```diff
@@ -0,0 +1,106 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver && test
+
+package kubeapiserver
+
+import (
+	"context"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDeploymentParser_Parse(t *testing.T) {
+	parser := newdeploymentParser()
+	deployment := &appsv1.Deployment{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "test-deployment",
+			Namespace: "test-namespace",
+			Labels:    map[string]string{"test-label": "test-value", "tags.datadoghq.com/env": "env", "tags.datadoghq.com/service": "service", "tags.datadoghq.com/version": "version"},
+		},
+	}
+	entity := parser.Parse(deployment)
+	storedDeployment, ok := entity.(*workloadmeta.KubernetesDeployment)
+	require.True(t, ok)
+	assert.IsType(t, &workloadmeta.KubernetesDeployment{}, entity)
```

Comment:
Is this testing the same as the assert just above?

Also, given that we're checking all the parameters of the deployment, I think it'd be good to define 	expectedDeployment := &workloadmeta.KubernetesDeployment{ ... } and have a single assert `assert.Equal(t, expectedDeployment, storedDeployment)`.

Reply:
Yep, indeed it makes sense